consensus/dummy/consensus.go

```diff
@@ -156,7 +156,7 @@ func (self *DummyEngine) verifyHeaderGasFields(config *params.ChainConfig, heade
 	if err != nil {
 		return fmt.Errorf("failed to calculate base fee: %w", err)
 	}
-	if !bytes.Equal(expectedRollupWindowBytes, header.Extra) {
+	if len(header.Extra) < len(expectedRollupWindowBytes) || !bytes.Equal(expectedRollupWindowBytes, header.Extra[:len(expectedRollupWindowBytes)]) {
```

Comment:
should we do this check only for the DUpgrade?

Reply:
Previously we required that the byte array was identical, now we are loosening it to require that the first 80 bytes are identical to ensure that the dynamic fee window byte array is correct.